### PR TITLE
Bech32 support

### PR DESCRIPTION
**Pull request by KotlinGeekDev**
The PR adds support for converting hex-encoded keys to their Bech32 representations

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -44,7 +44,7 @@ dependencies {
     implementation 'androidx.navigation:navigation-fragment-ktx:2.5.1'
     implementation 'androidx.navigation:navigation-ui-ktx:2.5.1'
     implementation project(path: ':nostrpostrlib')
-    implementation 'fr.acinq.secp256k1:secp256k1-kmp-jvm:0.6.4'
+    //implementation 'fr.acinq.secp256k1:secp256k1-kmp-jvm:0.6.4'
```
Comment (Giszmo) [inline app/build.gradle:47]: No commenting out, please. Just remove teh line if it's not needed anymore.

File: build.gradle [hunk 2]
```diff
@@ -9,8 +9,10 @@ buildscript {
     }
 }
 
+
+
+
```
Comment (Giszmo) [inline build.gradle:14]: Don't add empty lines like this, please. Standard formatting would allow up to one empty line between commands, zero at the end of blocks, mandate one empty line between functions, etc.

```suggestion
```

File: build.gradle [hunk 3]
```diff
@@ -9,8 +9,10 @@ buildscript {
...
 plugins {
-    id 'com.android.application' version '7.2.1' apply false
```
Comment (Giszmo) [inline build.gradle:13]: Does the `app` in the repo still start with this change?

File: examples/build.gradle [hunk 4]
```diff
@@ -6,6 +6,8 @@ java {
     targetCompatibility = JavaVersion.VERSION_11
 }
 
+
+
```
Comment (Giszmo) [inline examples/build.gradle:10]:
```suggestion
```

File: examples/build.gradle [hunk 5]
```diff
@@ -14,6 +16,8 @@ dependencies {
     implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
     implementation 'com.madgag.spongycastle:core:1.51.0.0'
     implementation 'com.madgag.spongycastle:prov:1.51.0.0'
+
+
```
Comment (Giszmo) [inline examples/build.gradle:20]: Generally, when making PRs, please look at your own PR and what the reviewer will see. The first impression is: 18 files change. Try to either functionally change files or not touch them at all. Or if you want to change or fix the formatting, fix it for all files.

```suggestion
```

File: nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+//        val version = data[0]
+//        require(version in 0..16) { "invalid segwit version" }
+//        val bin = five2eight(data, 1)
+//        require(bin.size in 2..40) { "invalid witness program length ${bin.size}" }
+//        if (version == 0.toByte()) require(encoding == Encoding.Bech32) { "version 0 must be encoded with Bech32" }
+//        if (version == 0.toByte()) require(bin.size == 20 || bin.size == 32) { "invalid witness program length ${bin.size}" }
+//        if (version != 0.toByte()) require(encoding == Encoding.Bech32m) { "version 1 to 16 must be encoded with Bech32m" }
+//        return Triple(hrp, version, bin)
+//    }
+
```
Comment (Giszmo) [inline nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt:252]: No commented code please.

File: nostrpostrlib/src/main/java/nostr/postr/Utils.kt [hunk 7]
```diff
@@ -24,8 +24,11 @@ object Utils {
     fun pubkeyCreate(privKey: ByteArray) =
         secp256k1.pubKeyCompress(secp256k1.pubkeyCreate(privKey)).copyOfRange(1, 33)
 
-    fun sign(data: ByteArray, privKey: ByteArray): ByteArray =
-        secp256k1.signSchnorr(data, privKey, null)
+    fun sign(data: ByteArray, privKey: ByteArray): ByteArray {
+        val randomness = ByteArray(32)
+        random.nextBytes(randomness)
```
Comment (Giszmo) [inline nostrpostrlib/src/main/java/nostr/postr/Utils.kt:29]: doesn't `secp256k1.signSchnorr()` do the same when called with `null` as randomness?

File: settings.gradle [hunk 8]
```diff
@@ -18,3 +18,4 @@ include ':nostrpostrlib'
 include ':bip-schnorr-kotlin'
 include ':examples'
 include ':NostrRelay'
+include ':nostrpostr-android'
```
Comment (Giszmo) [inline settings.gradle:21]: Why did you add this? It's basically empty? Why not use `app` that is there already? Please remove it or use it but don't just drop dead code in the project.

**Review by Giszmo — CHANGES_REQUESTED** [review]
Thanks for adding bech but please look into what you've done to the rest of the repo and undo what's not necessary.

Tests would also be welcome.

Please make sure `app` can still be launched.

**Comment by KotlinGeekDev** [comment]
Thanks for the review. I think it would be better to make another cleaner PR.

**Comment by Giszmo** [comment]
I don't mind showing [how the sausage got made](https://idioms.thefreedictionary.com/how+the+sausage+gets+made). Feel free to edit or close and open a new one.

**Comment by KotlinGeekDev** [comment]
:sweat_smile:
